plugins/git/commands

```diff
@@ -6,14 +6,13 @@ git_archive_all() {
   TMP_WORK_DIR=$(mktemp -d)
   chmod 755 $TMP_WORK_DIR
   unset GIT_DIR GIT_WORK_TREE
-  git clone -q $DOKKU_ROOT/$APP $TMP_WORK_DIR &> /dev/null
   pushd $TMP_WORK_DIR > /dev/null
+  git clone -q $DOKKU_ROOT/$APP $TMP_WORK_DIR &> /dev/null
   git config advice.detachedHead false
   git checkout $REV &> /dev/null
   git submodule update --init --recursive &> /dev/null
   find -name .git -prune -exec rm -rf {} \; > /dev/null
   tar c .
-  popd > /dev/null
```

Comment:
It is and that seems to be the issue. Try adding that line back without the `> /dev/null` and running `ps:rebuild <app>` from `~root/dokku`. It throws `/var/lib/dokku/plugins/git/commands: line 16: popd: /root/dokku: Permission denied`

Basically we `pushd $TMP_WORK_DIR` as the `dokku` user and that puts `~root/dokku` on the stack. Then `popd` as `dokku` fails to `cd` back to `~root/dokku`.